### PR TITLE
Fix #1091

### LINKED ISSUE #1091: Clients always open on first screen
Issue was originally reported [here](http://www.mail-archive.com/awesome@naquadah.org/msg08282.html), but I can reproduce it

Output of `awesome --version`:
```
awesome v3.5.2-1968-ge86ddaa-dirty (The Fox)
 • Compiled against Lua 5.2.4 (running with Lua 5.2)
 • D-Bus support: ✔
```

**How to reproduce the issue:**

- Move mouse pointer to some other screen than screen 1
- Open a terminal

**Actual result:**

Terminal opens on screen 1

**Expected result:**

Terminal opens on screen `awful.screen.focused()`

Looking at the diff of 21787444e4f1, this removes a call to `awful.screen.focused()`, but doesn't add one. Which part of the code is now responsible for assigning a default screen?

### PR DESCRIPTION
**Pull request by Elv13**
The test error will be easy enough to fix. Just disable the connect_sighnal("property::screen") in the first commit, then add a second commit to undo that and monkey-patch awful.screen.preferred to nil for the current tests, then put it back to its original value for the new tests we will need to prevent this from regressing. The connect_signal("property::screen") check is still necessary for the tests to be useful, as what they really test is that c.screen is set only once (to prevent the other use cases from regressing, again...). I don't have time for it now.

### CODE REVIEW

File: lib/awful/screen.lua [hunk 1]
```diff
@@ -213,6 +214,15 @@ function screen.object.set_padding(self, padding)
     end
 end
 
+--- Return the currently screen for new clients.
```
Comment (blueyed) [inline lib/awful/screen.lua]: s/currently/current/

File: lib/awful/screen.lua [hunk 2]
```diff
@@ -213,6 +214,15 @@ function screen.object.set_padding(self, padding)
...
+-- This is exactly the same as `awful.screen.focused` exept it avoids clients
```
Comment (blueyed) [inline lib/awful/screen.lua]: s/exept/except/

**Comment by blueyed** [comment]
Thanks!

**Comment by Elv13** [comment]
I will fix the build later. As said in the first comment, this commit break an assumption enforced by the multi-screen tests. Enforcing that assumption is still necessary to prevent your startup notification screen bug from regressing.

**Comment by blueyed** [comment]
> I will fix the build later.

Thanks!
I somehow oversaw that, and did not pay attention apparently: I did not want to break master.

**Comment by Elv13** [comment]
Don't worry, it isn't a bug, just a test failure due to the way the test are written. Master is not broken. Only the build is reported to have a failed minor test.

**Comment by blueyed** [comment]
yeah, that's what I've meant - it also runs fine for me.. :)

**Pull request closed without merging**